src/main/java/uk/gov/hmcts/divorce/systemupdate/event/SystemAttachScannedDocuments.java

```diff
@@ -42,18 +68,78 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
                 .done();
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+
+        final CaseData caseData = details.getData();
+
+        caseData.getDocuments().setScannedSubtypeReceived(null);
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(caseData)
+            .build();
+    }
+
     public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
         final CaseDetails<CaseData, State> details,
         final CaseDetails<CaseData, State> beforeDetails) {
+
         log.info(
             "[aboutToSubmit]: saving currentState({}) into caseData({}) to retain it post state transition",
             details.getState(), details.getId()
         );
-        var caseData = details.getData();
+
+        final CaseData caseData = details.getData();
+        final CaseData beforeCaseData = beforeDetails.getData();
         caseData.getApplication().setPreviousState(beforeDetails.getState());
 
+        final List<ListValue<ScannedDocument>> afterScannedDocs = caseData.getDocuments().getScannedDocuments();
+        final List<ListValue<ScannedDocument>> beforeScannedDocs =
+            isNotEmpty(beforeCaseData.getDocuments().getScannedDocuments())
+                ? beforeCaseData.getDocuments().getScannedDocuments()
+                : new ArrayList<>();
+
+        Optional<ScannedDocument> mostRecentScannedSubtypeReceived = Stream.ofNullable(afterScannedDocs)
+            .flatMap(Collection::stream)
+            .filter(element -> !beforeScannedDocs.contains(element))
+            .map(ListValue::getValue)
+            .filter(scannedDocument ->
+                EnumUtils.isValidEnum(CaseDocuments.ScannedDocumentSubtypes.class, scannedDocument.getSubtype().toUpperCase(Locale.ROOT))
+            )
+            .findFirst();
+
+        if (mostRecentScannedSubtypeReceived.isPresent()) {
+            final ScannedDocument scannedDocument = mostRecentScannedSubtypeReceived.get();
+            final CaseDocuments.ScannedDocumentSubtypes scannedDocumentSubtype =
+                CaseDocuments.ScannedDocumentSubtypes.valueOf(scannedDocument.getSubtype().toUpperCase(Locale.ROOT));
+            final DocumentType documentType = getDocumentType(scannedDocumentSubtype);
+
+            if (isEmpty(documentType)) {
+                return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                    .errors(singletonList("Scanned Document Subtype is an invalid form type"))
```

Comment:
I think we can't throw error while bulk scan attaches document to a case. What probably we should do is either classify it as Other or just attach it as it happens now ?